cnd/src/http_api/mod.rs

```diff
@@ -306,33 +309,31 @@ impl<'de> Deserialize<'de> for DialInformation {
 /// to another implementation. This allows us to keep the number of arguments to
 /// HTTP API controllers small and still access all the functionality we need.
 #[derive(Debug)]
-pub struct Dependencies<M, S, A, B, N> {
-    pub metadata_store: Arc<M>,
-    pub state_store: Arc<S>,
+pub struct Dependencies<A, B, S> {
+    pub metadata_store: Arc<InMemoryMetadataStore>,
```

Comment:
```suggestion
    pub metadata_store: Arc<MetadataStore>,
```
I understand that a MedatadaStore will always be in memory. As all Rust struct are :)

Reply:
Commented in another thread on this.  Perhaps we can chat offline about the motivation - basically I am not a big fan of generics when there is only, and only likely ever going to be, a single struct that implements the trait.  It seems to add needless complexity, of course you have much more Rust experience so I am open to your opinions on this.

Reply:
I don't understand your comment.

What I am saying is: Why do we call this `InMemoryMedataStore` when, as per your previous comments, will ever be only one type of `MetadataStore` due to the fact that we always load db items in this `InMemoryMedataStore`.

I am just proposing a rename here. Not sure you why you are going about generics and all.

I don't like the fact that we prefix the name with `InMemory`. If there is not going to be a `OnFilesystem` version of this then let's not prefix it with `InMemory`. Let's rename to `MetadataStore`.

Reply:
Cool, I agree with you totally.  I have it on my list of clean ups to remove the generics and just have a single MetatadataStore.  And do the same for StateStore.